modules/end-to-end/pages/building-tekton-tasks.adoc

```diff
@@ -0,0 +1,298 @@
+
+* A quay organization to be used with on-pr and push pipelines.
+
+* Public repositories within the quay organization to push the bundles to when using the on-pr and push pipelines.
+
+* A quay organization for the released bundles.
+
+* A public repository within the release organization for the trusted tasks list.
+
+* Access to the {ProductName} environment.
+
+* Necessary permissions to create and manage components in {ProductName}.
+
+
+== Onboarding Steps
+
+The onboarding of tasks to {ProductName} is similar to onboarding other applications/components to {ProductName}, with the exception of not having a Dockerfile (we are not building images).
+
+
+=== Application, Components and ImageRepository
+
+Tekton tasks are built as standard OCI artifacts in {ProductName}.
+
+The xref:building:/creating.adoc[Creating applications and components] page can be referenced for more information on how to create an Application for all tasks and a Component for each task within this Application, as well as an ImageRepository resource to be used within the quay organization.
+
+
+[NOTE] 
+====
+* Each task and task version in the application should be represented by a component, each component should contain the pipeline in use (*tekton-bundle-builder-oci-ta*) and the context to the task within the GH repository (*tasks/echo/0.1* for example).
+* The Component's naming convention is <task_name>-v<task_version> (echo-v01)
+====
+
+Once these resources are added, a PR with *on-pr* and *push* pipelineruns will be sent to your repository.
+
+[NOTE] 
+====
+* The *on-pr* pipelinerun will be created with the context we  have defined for the component and with *on-cel-expression* to run the pipelinerun only in case a change was detected in the component, for example:
+
+[source, yaml]
+----
+pipelinesascode.tekton.dev/on-cel-expression: |
+    event == "pull_request"
+    && target_branch == "main"
+    && (
+    "tasks/echo/0.1/***".pathChanged() || ".tekton/echo-v01-pull-request.yaml".pathChanged()
+    )
+----
+However, when *branch protection* is enabled in your GH repository it might prevent the pipelinerun to run when needed for the branch protection, in this case updating the *on-cel-expression* in the on-pr pipelinerun will solve the issue.
+[source, yaml]
+----
+pipelinesascode.tekton.dev/on-cel-expression: event == "pull_request" && target_branch
+----
+
+* The *Push* pipelinerun is created with a default on-cel-expression.
+For now we need to manually update it to run only if changes were made in the context, example:
+
+[source, yaml]
+----
+pipelinesascode.tekton.dev/on-cel-expression: |
+    event == "push"
+    && target_branch == "main"
+    && (
+    "tasks/echo/0.1/***".pathChanged() || ".tekton/echo-v01-push.yaml".pathChanged()
+      )
+----
+====
+
+=== Integration tests
+Follow xref:testing:integration/index.adoc[Integration tests] to add integration tests to the application and components.
```

Comment:
Is the Conforma integration test added by default? Are there any other recommendations for having an integration test?

If we are only going to mention the Conforma test, I feel like we can just make a page/section about that in the testing module and then xref from here. These are generally beneficial to have to enable policy-driven-development.

Reply:
In Konflux CI it is added by default and pass, I encountered some issues when it ran on downstream Konflux for the whole application.
Adding contexts and integration test per component solved it.
So I guess we can remove this section for upstream, but mentioned it for downstream, does it make sense?

Reply:
Sure. If we had a section on gitops for configuring (i.e. a pattern), then we could add it there... and then we could further augment that for downstream usage, potentially.